torch/distributed/distributed_c10d.py

```diff
@@ -2476,6 +2476,57 @@ def barrier(group=GroupMember.WORLD,
     else:
         work.wait()
 
+def monitored_barrier(group=GroupMember.WORLD, timeout=None):
+
```

Comment:
nit: could explicitly say that this barrier could be inserted before collective calls to check whether some ranks are desynced